code/_onclick/hud/new_player.dm

```diff
@@ -348,14 +369,139 @@
 		set_button_status(FALSE)
 		return
 
-/atom/movable/screen/lobby/button/poll/update_overlays()
+/atom/movable/screen/lobby/button/bottom/poll/update_overlays()
 	. = ..()
 	if(new_poll)
 		. += mutable_appearance('icons/hud/lobby/poll_overlay.dmi', "new_poll")
 
-/atom/movable/screen/lobby/button/poll/Click(location, control, params)
+/atom/movable/screen/lobby/button/bottom/poll/Click(location, control, params)
 	. = ..()
 	if(!.)
 		return
 	var/mob/dead/new_player/new_player = hud.mymob
 	new_player.handle_player_polling()
+
+#define SHUTTER_MOVEMENT_DURATION 0.4 SECONDS
+#define SHUTTER_WAIT_DURATION 0.2 SECONDS
+
+/atom/movable/screen/lobby/button/collapse
+	name = "Collapse Lobby Menu"
+	icon = 'icons/hud/lobby/collapse_expand.dmi'
+	icon_state = "collapse"
+	base_icon_state = "collapse"
+	layer = LOBBY_COLLAPSE_BUTTON_LAYER
+	screen_loc = "TOP:-82,CENTER:-54"
+	always_shown = TRUE
+	var/matrix/ce_button_matrix = matrix()
+
+/atom/movable/screen/lobby/button/collapse/Click(location, control, params)
+	. = ..()
+	if(!.)
+		return
+	var/mob/dead/new_player/new_player = hud.mymob
+	//whether we are currently showing the menu to the client or not
+	var/inventory_status = hud.inventory_shown
+	base_icon_state = inventory_status ? "expand" : "collapse"
+	name = "[inventory_status ? "Collapse" : "Expand"] Lobby Menu"
+	set_button_status(FALSE)
+	//re-enable clicking the button when the shutter animation finishes
+	addtimer(CALLBACK(src, PROC_REF(set_button_status), TRUE), (2 * SHUTTER_MOVEMENT_DURATION + SHUTTER_WAIT_DURATION))
```

Comment:
this should be a client timer so it's more responsive under load. use the TIMER_CLIENT_TIME flag

Reply:
https://github.com/tgstation/tgstation/pull/76443/commits/e74765b3ea02e603b27f346dc73142c28b13f527